creusot/src/validate_terminates.rs

```diff
@@ -102,7 +105,7 @@ pub(crate) fn validate_terminates(ctx: &mut TranslationCtx) {
     }
 
     // detect mutual recursion
-    let cycles = petgraph::algo::kosaraju_scc(&call_graph);
+    let cycles = petgraph::algo::tarjan_scc(&call_graph);
```

Comment:
Not much. Tarjan algorithm is usually preferred, because it has a lower memory footprint, and is probably more efficient too (it requires only one traversal of the graph).